src/grt/src/MakeWireParasitics.cpp

```diff
@@ -74,6 +75,32 @@ MakeWireParasitics::MakeWireParasitics(utl::Logger* logger,
 {
 }
 
+MakeWireParasitics::MakeWireParasitics(utl::Logger* logger,
+                                       rsz::Resizer* resizer,
+                                       rsz::SpefWriter* spef_writer,
+                                       sta::dbSta* sta,
+                                       odb::dbTech* tech,
+                                       odb::dbBlock* block,
+                                       GlobalRouter* grouter)
+    : grouter_(grouter),
+      tech_(tech),
+      block_(block),
+      logger_(logger),
+      resizer_(resizer),
+      spef_writer_(spef_writer),
+      sta_(sta),
+      network_(sta_->getDbNetwork()),
+      parasitics_(sta_->parasitics()),
+      arc_delay_calc_(sta_->arcDelayCalc()),
+      min_max_(sta::MinMax::max()),
+      resistor_id_(1)
+{
+  for (sta::Corner* corner : *sta_->corners()) {
```

Comment:
Why creating a new constructor? Also, I don't think the class constructor should be writing anything related to the spef file.